### PR TITLE
Update service/client request/response API error returns

### PR DESCRIPTION
**Pull request by Lobotuerk**
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

**Comment by Lobotuerk** [comment]
CI job up to `rcl` and `test_rmw_implementation`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12456/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7423/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10167/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12375/)

File: rmw_fastrtps_shared_cpp/src/rmw_request.cpp [hunk 1]
```diff
@@ -38,16 +39,18 @@ __rmw_send_request(
...
 
-  if (client->implementation_identifier != identifier) {
-    RMW_SET_ERROR_MSG("node handle not from this implementation");
-    return RMW_RET_ERROR;
-  }
+  RMW_CHECK_ARGUMENT_FOR_NULL(ros_request, RMW_RET_INVALID_ARGUMENT);
+
+  RMW_CHECK_ARGUMENT_FOR_NULL(sequence_id, RMW_RET_INVALID_ARGUMENT);
+
+  rmw_ret_t returnedValue = RMW_RET_ERROR;
```
Comment (hidmic) [inline rmw_fastrtps_shared_cpp/src/rmw_request.cpp:53]: @Lobotuerk nit: consider removing all unnecessary blank lines above.

**Review by hidmic — APPROVED** [review]

File: rmw_fastrtps_shared_cpp/src/rmw_request.cpp (hunk 1 above)
Comment (Lobotuerk) [reply rmw_fastrtps_shared_cpp/src/rmw_request.cpp:50]: @hidmic 8f130e5 like this?

**Review by hidmic — APPROVED** [review]

**Comment by Lobotuerk** [comment]
CI up to `rcl` and `test_rmw_implementation`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12503/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7471/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10215/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12423/)

File: rmw_fastrtps_shared_cpp/src/rmw_request.cpp [hunk 2]
```diff
@@ -38,17 +39,16 @@ __rmw_send_request(
   const void * ros_request,
   int64_t * sequence_id)
 {
-  assert(client);
-  assert(ros_request);
-  assert(sequence_id);
+  RMW_CHECK_ARGUMENT_FOR_NULL(client, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    client handle,
```
Comment (Blast545) [inline rmw_fastrtps_shared_cpp/src/rmw_request.cpp:44]: Was this left purposely ?

**Review by Blast545 — APPROVED** [review]

File: rmw_fastrtps_shared_cpp/src/rmw_request.cpp (hunk 2 above)
Comment (Lobotuerk) [reply rmw_fastrtps_shared_cpp/src/rmw_request.cpp]: Updated on 486d9d8

**Comment by Lobotuerk** [comment]
CI after reviews, up to `rcl` and `test_rmw_implementation`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12510/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7478/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10222/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12430/)

**Comment by Lobotuerk** [comment]
All CI green, merging

**Pull request merged**